### PR TITLE
Import GP and contigs fixes (and API)

### PR DESCRIPTION
**Pull request by mberacochea**
This PR contains the changes to adjust the commands:
- import_summary .paths.gprops
- import_contigs

## Genome properties
Changed the Mongo model to store the values YES|PARTIAL|NO. Webclient PR https://github.com/EBI-Metagenomics/ebi-metagenomics-client/pull/137/files

## Contigs

For file indexes created using samtools or tabix use `.bgz` extension insted of `.gz`, they are compatible but not the same (http://www.htslib.org/doc/bgzip.html)

### fasta

The API now uses the compressed version of the fasta and the fasta index file.
Compressed bgzip (part of tabix package) files produce 2 index files:
```
ERZ782882_FASTA.fasta.bgz
ERZ782882_FASTA.fasta.bgz.fai
ERZ782882_FASTA.fasta.bgz.gzi
```
Command: 
```bash
$ bgzip < ERZ782882_FASTA.fasta > ERZ782882_FASTA.fasta.bgz && samtools faidx ERZ782882_FASTA.fasta.bgz
```

### functional-annotations

Drop `FILE.**contigs**.annotations.gff.*`. Example
```
ERZ782882_FASTA.contigs.annotations.gff => ERZ782882_FASTA.annotations.gff
```
Use `.bgz` instead of `.gz`.
```
ERZ782882_FASTA.annotations.gff.gz => ERZ782882_FASTA.annotations.gff.bgz
ERZ782882_FASTA.annotations.gff.gz.tbi => ERZ782882_FASTA.annotations.gff.bgz.tbi
```
### pathways-systems

Use `.bgz` instead of `.gz`. Example:
```
ERZ782882_FASTA.antismash.gff.gz => ERZ782882_FASTA.antismash.gff.bgz
ERZ782882_FASTA.antismash.gff.gz.tbi => ERZ782882_FASTA.antismash.gff.bgz.tbi
```

### CODE REVIEW

**Comment by KeteSakharova** [comment]
Do I need to fix pipeline then?
1) add: ERZ782882_FASTA.fasta.bgz.gzi
2) rename gff to ERZ782882_FASTA.annotations.gff
3) ERZ782882_FASTA.antismash.gff.gz => ERZ782882_FASTA.antismash.gff.bgz
4) ERZ782882_FASTA.antismash.gff.gz.tbi => ERZ782882_FASTA.antismash.gff.bgz.tbi

**Comment by mberacochea** [comment]
Changes needed in the pipeline:

1. add: ERZ782882_FASTA.fasta.bgz.gzi
bgzip the fasta file and then index with samtools faidx. This will generate the 2 index files required.

2. rename gff to ERZ782882_FASTA.annotations.gff
Yes.

3. ERZ782882_FASTA.antismash.gff.gz => ERZ782882_FASTA.antismash.gff.bgz
Yes.

4. ERZ782882_FASTA.antismash.gff.gz.tbi => ERZ782882_FASTA.antismash.gff.bgz.tbi
This will be automatically done with 3.

File: emgapianns/management/commands/import_summary.py [hunk 1]
```diff
@@ -42,43 +51,31 @@ def _check_source_file(source_file):
...
-                    logger.info('Could not find: %s' % source_file)
-                else:
-                    logger.info('Found: %s' % source_file)
-                    self._parse_and_load_summary_file(source_file, obj)
-                    break
-        elif self.suffix in self.kegg_pathway_suffix:
-            name = '%s.summary%s' % (obj.input_file_name, self.suffix)
-            source_file = os.path.join(rootpath, obj.result_directory, self.PATHWAY_SUB_DIR, name)
+            name = "%s_summary%s" % (obj.input_file_name, self.suffix)
+            source_file = os.path.join(rootpath, obj.result_directory, name)
```
Comment (mscheremetjew) [inline emgapianns/management/commands/import_summary.py:70]: The way you simplified this assumes that obj.result_directory resolves the correct sub directory path, which think it doesn't.
For instance the antismash GFF file resolves to the following sub directory: ERZXXXXXX_FASTA/pathways-systems

File: emgapianns/management/commands/import_summary.py [hunk 2]
```diff
@@ -14,20 +27,16 @@
...
 
-    FUNCTION_SUB_DIR = 'functional-annotation'
-    PATHWAY_SUB_DIR = 'pathways-systems'
-
     def __init__(self):
         super().__init__()
-        self.suffixes = ['.ips', '.ipr', '.go', '.go_slim', '.pfam', '.ko', '.gprops', '.antismash']
-        self.pathway_suffixes = ['.gprops', '.antismash', '.kegg_pathways']
-        self.kegg_pathway_suffix = ['.kegg_pathways']
+        self.suffixes = [".ipr", ".go", ".go_slim", ".pfam", ".ko", ".paths.gprops", ".antismash"]
```
Comment (mscheremetjew) [inline emgapianns/management/commands/import_summary.py:32]: The format of the InterProScan summary file change completely between version 4 and 5. ipr will support v4 and ips v5 functionality. I cannot see that functionality anymore?

File: emgapianns/management/commands/import_summary.py (hunk 1 above)
Comment (mberacochea) [reply emgapianns/management/commands/import_summary.py:70]: This is a mistake, I accidentally overwrote the dependency-version-upgrades changes. I'll restore the file and add the changes from genome properties.

File: emgapianns/management/webuploader_configs/v5_0/assembly.json [hunk 3]
```diff
@@ -777,6 +777,98 @@
...
+      "group_type": null,
+      "real_name": "{}.annotations.gff.bgz.tbi",
+      "subdir": "functional-annotation"
+    },
+    {
+      "_accession_substitution": "full",
+      "_chunked": false,
+      "_required": true,
+      "downloadable": true,
+      "alias": null,
```
Comment (mscheremetjew) [inline emgapianns/management/webuploader_configs/v5_0/assembly.json:851]: `"alias": "{}_antismash.gff.bgz",`

**Pull request merged**